src/model/mmodule.nit

```diff
@@ -186,11 +186,16 @@ class MModule
 	end
 
 	# Register the imported modules (ie "import some_module")
+	# In the same time it register the imported package
 	# The visibility must be set with `set_visibility_for`.
 	fun set_imported_mmodules(imported_mmodules: Array[MModule])
 	do
 		for m in imported_mmodules do
 			self.model.mmodule_importation_hierarchy.add_edge(self, m)
+			if self.mpackage isa MPackage and m.mpackage isa MPackage then
```

Comment:
how could mpackage not be a MPackage? Or do you mean `!= null` instead of `isa`?

Reply:
Yes, it was to perform the null test. I changed it by using local variables and adaptive typing as you indicated.